client/post-editor/editor-sharing/accordion.jsx

```diff
@@ -82,18 +83,23 @@ const EditorSharingAccordion = React.createClass( {
 	},
 
 	hideSharing: function() {
-		return this.props.site &&
-			this.props.site.jetpack &&
-			! this.props.site.isModuleActive( 'publicize' ) &&
-			! this.props.site.isModuleActive( 'sharedaddy' ) &&
-			! this.props.site.isModuleActive( 'likes' );
+		return ( this.props.site &&
+				this.props.site.jetpack &&
+				! this.props.site.isModuleActive( 'publicize' ) &&
+				! this.props.site.isModuleActive( 'sharedaddy' ) &&
+				! this.props.site.isModuleActive( 'likes' ) ) ||
+			( this.props.site &&
+				this.props.site.jetpack &&
+				this.props.postType === 'page' &&
+				! this.props.site.isModuleActive( 'sharedaddy' ) &&
+				! this.props.site.isModuleActive( 'likes' ) );
```

Comment:
Adding to @nylen 's remarks, we should also consider using the state selector [`isJetpackModuleActive`](https://github.com/Automattic/wp-calypso/blob/37d3a88e3732ec131ad518eb0266ed076855e927/client/state/sites/selectors.js#L134-L151) in place of the site instance function, though this is more a general refactoring remark.

@nylen it appears you're missing the `page` type consideration in your alternative approach.